crates/bevy_render/src/texture/hdr_texture_loader.rs

```diff
@@ -47,17 +48,9 @@ impl AssetLoader for HdrTextureLoader {
             reader.read_to_end(&mut bytes).await?;
             let decoder = image::codecs::hdr::HdrDecoder::new(bytes.as_slice())?;
             let info = decoder.metadata();
-            let rgb_data = decoder.read_image_hdr()?;
-            let mut rgba_data = Vec::with_capacity(rgb_data.len() * format.pixel_size());
-
-            for rgb in rgb_data {
-                let alpha = 1.0f32;
-
-                rgba_data.extend_from_slice(&rgb.0[0].to_ne_bytes());
-                rgba_data.extend_from_slice(&rgb.0[1].to_ne_bytes());
-                rgba_data.extend_from_slice(&rgb.0[2].to_ne_bytes());
-                rgba_data.extend_from_slice(&alpha.to_ne_bytes());
-            }
+            let buffer_size = decoder.total_bytes().try_into().unwrap();
```

Comment:
Now that I see this again, it might be better to introduce an "out of bounds" Error instead of panicking. The only error that could happen here, AFAICS, is that the u64 to usize cast exceeds usize::MAX.